daemon/src/maker_cfd_actor.rs

```diff
@@ -2,281 +2,410 @@ use crate::db::{
-                            None => {
-                                contract_setup_message_buffer.push(msg);
-                                continue;
-                            }
-                            Some(inbox) => inbox,
-                        };
-
-                        inbox.send(msg).unwrap();
-                    }
-                    maker_cfd_actor::Command::CfdSetupCompleted { order_id, dlc } => {
-                        println!("Setup complete, publishing on chain now...");
-
-                        current_contract_setup = None;
-                        contract_setup_message_buffer = vec![];
-
-                        insert_new_cfd_state_by_order_id(
-                            order_id,
-                            CfdState::PendingOpen {
-                                common: CfdStateCommon {
-                                    transition_timestamp: SystemTime::now(),
-                                },
-                                dlc: dlc.clone(),
-                            },
-                            &mut conn,
-                        )
-                        .await
-                        .unwrap();
-
-                        cfd_feed_actor_inbox
-                            .send(load_all_cfds(&mut conn).await.unwrap())
-                            .unwrap();
-
-                        let txid = wallet.try_broadcast_transaction(dlc.lock).await.unwrap();
-
-                        println!("Lock transaction published with txid {}", txid);
-
-                        // TODO: tx monitoring, once confirmed with x blocks transition the Cfd to
-                        // Open
-                    }
+                        command: TakerCommand::OutProtocolMsg { setup_msg: msg },
+                    });
                 }
+            },
+            setup_contract_actor::OwnParams::Maker(maker_params),
+            sk,
+            self.oracle_pk,
+            cfd,
+            self.wallet.clone(),
+        );
+
+        self.current_contract_setup.replace(inbox.clone());
+
+        for msg in self.contract_setup_message_buffer.drain(..) {
+            inbox.send(msg).unwrap();
+        }
+
+        // TODO: Should we do this here or already earlier or after the spawn?
+        insert_new_cfd_state_by_order_id(
+            order_id,
+            CfdState::ContractSetup {
+                common: CfdStateCommon {
+                    transition_timestamp: SystemTime::now(),
+                },
+            },
+            &mut conn,
+        )
+        .await?;
+        self.cfd_feed_actor_inbox
+            .send(load_all_cfds(&mut conn).await?)?;
+
+        let address = ctx
+            .address()
+            .expect("actor to be able to give address to itself");
+        address
+            .do_send_async(CfdSetupCompleted {
+                order_id,
+                dlc: actor.await,
+            })
+            .await?;
```

Comment:
The `actor.await` needs to be in a separate `tokio::spawn`, otherwise we have previously mentioned deadlock! :)

Reply:
like this ? 

```
        tokio::spawn(address.do_send_async(CfdSetupCompleted {
            order_id,
            dlc: actor.await,
        }));
```

Reply:
```rust
tokio::spawn(async move {
	address.do_send_async(CfdSetupCompleted {
       order_id,
       dlc: actor.await,
    })
});
```